grid.py

```diff
@@ -114,6 +116,29 @@ def get_mask_invalid_actions(self, state=None, done=None):
                     break
         return mask
 
+    def get_backward_mask(self, state=None, done=None, obs2state=False):
+        """
+        Returns a vector of length the action space + 1: True if backward action is invalid
+        given the current state, False otherwise.
+        """
+        if state is None:
+            state = self.state.copy()
+        if done is None:
+            done = self.done
+        if done:
+            mask = [True for _ in range(len(self.action_space))]
+            mask.append(False)
+            return mask
+        if obs2state == True:
+            state = self.obs2state(state.cpu().numpy())
+        mask = [False for _ in range(len(self.action_space) + 1)]
+        for idx, a in enumerate(self.action_space):
+            for d in a:
+                if state[d] - 1 < 0:
+                    mask[idx] = True
+                    break
```

Comment:
Just for the records and for potential discussion if necessary, this will always make backward action eos (last in action space) valid, which I would argue is not correct. In any case, this function will not be necessary.